src/lib/y2packager/dialogs/addon_selector.rb

```diff
@@ -1,37 +1,40 @@
-# ------------------------------------------------------------------------------
-# Copyright (c) 2017 SUSE LLC, All Rights Reserved.
+# Copyright (c) [2017-2020] SUSE LLC
 #
-# This program is free software; you can redistribute it and/or modify it under
-# the terms of version 2 of the GNU General Public License as published by the
-# Free Software Foundation.
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
 #
 # This program is distributed in the hope that it will be useful, but WITHOUT
-# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
-# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
-# ------------------------------------------------------------------------------
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
 
 require "yast"
-require "erb"
-require "ui/installation_dialog"
+require "cwm/dialog"
 require "y2packager/resolvable"
+require "y2packager/widgets/addons_selector"
 
 Yast.import "AddOnProduct"
 Yast.import "Mode"
 Yast.import "ProductFeatures"
-Yast.import "Report"
 Yast.import "Stage"
-Yast.import "UI"
 Yast.import "Wizard"
 
 module Y2Packager
   module Dialogs
     # Dialog which shows the user available products on the medium
-    class AddonSelector < ::UI::InstallationDialog
-      include Yast::Logger
-      include ERB::Util
-
+    class AddonSelector < ::CWM::Dialog
```

Comment:
**NP** (question): It's not clear to me if it worth it to move and rename both, the dialog and the widget, to a more semantic name. I mean

* Dialog, from `AddonSelector` to `ProductsSelection`
* Widget, from `AddonsSelector` to `ProductsSelector`
* Things named `addon(s)` to `product(s)`

So far, we are managing here three different (?) concepts

* Addon
* Product
* ProductLocation

and it's a bit confusing * *to me**. That's why I decided to ask first and change them by now, but IMO it could be helps to understand the code and differences, if any, with the addons selection in yast2-registration.

Reply:
We already have the [Installation::Widgets::ProductSelector](https://github.com/yast/yast-installation/blob/master/src/lib/installation/widgets/product_selector.rb) and the [Installation::Dialogs::ProductSelection](https://github.com/yast/yast-installation/blob/master/src/lib/installation/dialogs/product_selection.rb) classes.

These are for selecting the base product in the first installation dialog. I think the `Addon*` names are OK as they refer to modules and extensions (not a base product) so this should avoid confusion with the classes above.

And to the other concepts: yes, that can be confusing, but again this happened because they implement different things. E.g. that `ProductLocation` represents the content of the `/media.1/products` file on the medium, that something different than the libzypp products...